clash-lib/src/Clash/Core/Type.hs

```diff
@@ -596,6 +630,11 @@ isTypeFamilyApplication tcm (tyView -> TyConApp tcNm _args)
   | Just (FunTyCon {}) <- lookupUniqMap tcNm tcm = True
 isTypeFamilyApplication _tcm _type = False
 
+argView :: TyConMap -> Type -> Type
+argView m t = case reduceTypeFamily m t of
+  Nothing -> t
+  Just tR -> argView m tR
```

Comment:
Maybe `reduceTypeFamily1` + `reduceTypeFamily`, like `coreView` / `coreView1`?

Reply:
It's not exported, and only used by `reduceTypeFamily`